src/sage/combinat/posets/posets.py

```diff
@@ -8945,8 +8945,8 @@ def __iter__(self):
             sage: list(P)
             [Finite poset containing 2 elements, Finite poset containing 2 elements]
         """
-        from sage.graphs.digraph_generators import DiGraphGenerators
-        for dig in DiGraphGenerators()(self._n, is_poset):
+        from sage.graphs.digraph_generators import DiGraphGenerators as DG
+        for dig in DG().nauty_posetg(f"{self._n} o"):
```

Comment:
I would prefer not to use `as DG`, because it makes code much harder to read for me personally.  Also, the import could actually go to the module level, no?